gps2val/gpsd_feeder.py

```diff
@@ -83,37 +83,32 @@ def __init__(self, config, consumer):
 
     def loop(self):
         print("gpsd receive loop started")
-        while self.running:
-            try:
-                if self.gpsd.waiting():
-                    report = self.gpsd.next()
-                    if report['class'] == 'TPV':
-                        print("")
-                        self.collecteddata['Vehicle.CurrentLocation.Latitude']= getattr(report,'lat',None)
-                        self.collecteddata['Vehicle.CurrentLocation.Longitude']= getattr(report,'lon',None)
-                        self.collecteddata['Vehicle.CurrentLocation.Altitude']= getattr(report,'alt', None)
-                        self.collecteddata['Vehicle.Speed']= getattr(report,'speed',None)
-                        self.collecteddata['Vehicle.CurrentLocation.Timestamp']= getattr(report,'time',None)
-                        self.collecteddata['Vehicle.CurrentLocation.Heading']= getattr(report,'track',None)
-                        self.collecteddata['Vehicle.CurrentLocation.HorizontalAccuracy']= getattr(report,'eph',None)
-                        self.collecteddata['Vehicle.CurrentLocation.VerticalAccuracy']= getattr(report,'epv',None)
-
-                        self.consumer.setData(self.collecteddata)
-                time.sleep(self.interval) 
-            except Exception as e:
-                print("Got exception: ")
-                print(e)
-                time.sleep(self.interval) 
-                continue
+         for result in self.client.dict_stream(filter=["TPV"]):
+            if self.running:
+                print("")
+                self.collecteddata['Vehicle.CurrentLocation.Latitude']= result.get('lat',None)
+                self.collecteddata['Vehicle.CurrentLocation.Longitude']= result.get('lon',None)
+                self.collecteddata['Vehicle.CurrentLocation.Altitude']= result.get('alt', None)
+                self.collecteddata['Vehicle.Speed']= result.get('speed',None)
+                self.collecteddata['Vehicle.CurrentLocation.Timestamp']= result.get('time',None)
+                self.collecteddata['Vehicle.CurrentLocation.Heading']= result.get('track',None)
+                self.collecteddata['Vehicle.CurrentLocation.HorizontalAccuracy']= result.get('eph',None)
+                self.collecteddata['Vehicle.CurrentLocation.VerticalAccuracy']= result.get('epv',None)
+
+                self.consumer.setData(self.collecteddata)
+                time.sleep(self.interval)
+            else:
+                break
 
      
 
 
     def shutdown(self):
-        self.running=False
-        self.consumer.shutdown()
-        self.gpsd.close()
+        print("KUKSA client shutdown")
```

Comment:
I don't see where you shut down the kuksa client.
I think you need to resurrect the removed `self.consumer.shutdown()` line.

Reply:
Must've happened while rebase should'nt be deleted. You're right.